packages/amazon-cognito-identity-js/__tests__/StorageHelperRN.test.js

```diff
@@ -0,0 +1,126 @@
+describe('React native storage helper unit tests', () => {
+	describe('Accessor methods', () => {
+		beforeEach(() => {
+			jest.resetModules();
+		});
+
+		var items = {};
+		jest.mock('react-native', () => ({
+			AsyncStorage: {
+				setItem: jest.fn((item, value) => {
+					return new Promise((resolve, reject) => {
+						items[item] = value;
+						resolve(value);
+					});
+				}),
+				getItem: jest.fn((item, value) => {
+					return new Promise((resolve, reject) => {
+						resolve(items[item]);
+					});
+				}),
+				removeItem: jest.fn(item => {
+					return new Promise((resolve, reject) => {
+						resolve(delete items[item]);
+					});
+				}),
+			},
+		}));
+
+		const StorageHelper = require('../src/StorageHelper-rn.js').default;
+		const storageHelper = new StorageHelper();
+		test('setting an item', () => {
+			expect(
+				storageHelper.getStorage().setItem('testKey', 'testValue')
+			).toEqual('testValue');
+		});
+		test('getting an item', () => {
+			expect(storageHelper.getStorage().getItem('testKey')).toEqual(
+				'testValue'
+			);
+		});
+		test('removing an item', () => {
+			expect(storageHelper.getStorage().removeItem('testKey')).toEqual(true);
+		});
+		test('clearing the storage', () => {
+			storageHelper.getStorage().setItem('item1', 'value1');
+			expect(storageHelper.getStorage().clear()).toEqual({});
+		});
+	});
+
+	describe('Testing StorageHelper sync', () => {
+		beforeEach(() => {
+			jest.resetModules();
+		});
+		test('Happy case for syncing', () => {
+			var items = { key: 'value1', key2: 'value2' };
+			jest.mock('react-native', () => ({
+				AsyncStorage: {
+					setItem: jest.fn((item, value) => {
+						return new Promise((resolve, reject) => {
+							items[item] = value;
+							resolve(value);
+						});
+					}),
+					getAllKeys: jest.fn(callback => {
+						callback(null, Object.keys(items));
+					}),
+					multiGet: jest.fn((keys, callback) => {
+						const values = keys.map(key => [key, items[key] || null]);
+						callback && callback(null, values);
+
+						return values;
+					}),
+				},
+			}));
+
+			const StorageHelper = require('../src/StorageHelper-rn.js').default;
+			const storageHelper = new StorageHelper();
+			const callback = jest.fn();
+
+			//localStorage would then be out of sync with AsyncStorage
+			storageHelper.getStorage().setItem('key', 'value');
+			storageHelper.getStorage().sync(callback);
+			expect(callback.mock.calls[0][1]).toEqual('SUCCESS');
```

Comment:
I think here we should first check that AsyncStorage and MemoryStorage are actually out of sync by attempting to get a value by one of those keys and expect it to be undefined. Then perform the sync and get it again, this time expecting it to equal the value.
```suggestion
			// MemoryStorage would then be out of sync with AsyncStorage
			expect(storageHelper.getStorage().getItem('key')).toBeUndefined();
			storageHelper.getStorage().sync(callback);
			expect(callback.mock.calls[0][1]).toEqual('SUCCESS');
			expect(storageHelper.getStorage().getItem('key')).toEqual('value1');
```

Reply:
```
// MemoryStorage would then be out of sync with AsyncStorage
expect(storageHelper.getStorage().getItem('key')).toBeUndefined();
storageHelper.getStorage().setItem('key', 'value1');
storageHelper.getStorage().sync(callback);
expect(callback.mock.calls[0][1]).toEqual('SUCCESS');
expect(storageHelper.getStorage().getItem('key')).toEqual('value1');
```